onnxruntime/lora/adapter_format_utils.cc

```diff
@@ -75,35 +77,82 @@ const Adapter* ValidateAndGetAdapterFromBytes(gsl::span<const uint8_t> bytes) {
+
+    InlinedVector<uint8_t> be_data(data.size());
+    auto status = disp.InvokeRet<Status, WriteDataForLittleEndian>(data, be_data);
+    ORT_THROW_IF_ERROR(status);
+    data_vec = flat_builder.CreateVector<uint8_t>(be_data.data(), be_data.size());
+  } else {
+    data_vec = flat_builder.CreateVector(data.data(), data.size());
+  }
   fbs_tensor = CreateParameter(flat_builder, name_str, shape_vec, data_type, data_vec);
 }
 
+template <class T>
+struct ReadDataForBigEndian {
+  Status operator()(gsl::span<const uint8_t> src, Tensor& dst) const {
+    auto src_span = ReinterpretAsSpan<const unsigned char>(src);
+    auto dst_span = dst.MutableDataAsSpan<T>();
+    return onnxruntime::utils::ReadLittleEndian<T>(src_span, dst_span);
+  }
+};
+
 std::pair<std::string, OrtValue> CreateOrtValueOverLoraParameter(const Parameter& param) {
   OrtValue result;
 
   std::string name;
   LoadStringFromLoraFormat(name, param.name());
 
   const auto data_type = param.data_type();
-  gsl::span<const int64_t> shape_span(param.dims()->data(), param.dims()->size());
-
-  static const OrtMemoryInfo cpu_meminfo(CPU, OrtAllocatorType::OrtDeviceAllocator);
-
-  auto elem_type = DataTypeImpl::TensorTypeFromONNXEnum(static_cast<int32_t>(data_type))->GetElementType();
-  // const_cast is necessery due to Tensor class API
-  Tensor::InitOrtValue(elem_type,
-                       TensorShape(shape_span),
-                       const_cast<uint8_t*>(param.raw_data()->data()),
-                       cpu_meminfo,
-                       result);
+  // Copying shape takes care of endianess using flatbuffers accessors
+  TensorShapeVector shape(param.dims()->begin(), param.dims()->end());
+  static const AllocatorPtr cpu_allocator = std::make_shared<CPUAllocator>();
+  const auto elem_type = DataTypeImpl::TensorTypeFromONNXEnum(static_cast<int32_t>(data_type))->GetElementType();
+
+  // If BE, we a allocate memory within the tensor and copy there swapping bytes
+  if constexpr (endian::native == endian::big) {
+    gsl::span<const uint8_t> src_span(param.raw_data()->data(), param.raw_data()->size());
+    Tensor tensor(elem_type, shape, cpu_allocator);
+    onnxruntime::utils::MLTypeCallDispatcher<float, double, int8_t, uint8_t,
+                                             int16_t, uint16_t, int32_t, uint32_t,
+                                             int64_t, uint64_t,
+                                             BFloat16, MLFloat16>
+        disp(static_cast<int32_t>(data_type));
+
+    auto status = disp.InvokeRet<Status, ReadDataForBigEndian>(src_span, tensor);
+    ORT_THROW_IF_ERROR(status);
+    Tensor::InitOrtValue(std::move(tensor), result);
+  } else {
+    // const_cast is necessary due to Tensor class API
+    Tensor::InitOrtValue(elem_type,
+                         TensorShape(shape),
+                         const_cast<uint8_t*>(param.raw_data()->data()),
+                         cpu_allocator->Info(),
```

Comment:
Can we avoid creating the local cpu_allocator on the LE path and use cpu_meminfo here?